src/NRediSearch/Client.cs

```diff
@@ -458,6 +547,7 @@ public Task<bool> ReplaceDocumentAsync(string docId, Dictionary<string, RedisVal
         /// <param name="score">the document's index score, between 0 and 1</param>
         /// <param name="replace">if set, and the document already exists, we reindex and update it</param>
         /// <returns>true on success</returns>
+        /// [Deprecated] Use IDatabase.HashSet instead.
```

Comment:
in C#, you can use `ObsoleteAttribute` for this, and it will be advertised to the caller, for example:

``` c#
/// <returns>true on success</returns>
[Obsolete("Use IDatabase.HashSet instead.")]
public bool AddHash(RedisKey docId, double score, bool replace)
```

Note that there is an optional boolean you can add, which determines whether it is a warning or a hard error. Since the old code remains, I'm guessing leaving it as the default (warning) is fine.

Reply:
Should I remove the `[Deprecated]` from the docs? I see that in some places in the code we do use it

Reply:
if there is an `[Obsolete]`, then it is self-evident and doesn't **need** additional remarks, but those remarks won't do any harm either; either way is fine